lib/money/money_parser.rb

```diff
@@ -1,56 +1,140 @@
+# frozen_string_literal: true
+
 # Parse an amount from a string
 class MoneyParser
-  MARKS = "\.,·’"
-  EXTRA_MARKS = "\s˙'"
+  class MoneyFormatError < ArgumentError; end
+
+  MARKS = %w[. , · ’ ˙ '] + [' ']
+
+  ESCAPED_MARKS = Regexp.escape(MARKS.join)
+  ESCAPED_NON_SPACE_MARKS = Regexp.escape((MARKS - [' ']).join)
+  ESCAPED_NON_DOT_MARKS = Regexp.escape((MARKS - ['.']).join)
+  ESCAPED_NON_COMMA_MARKS = Regexp.escape((MARKS - [',']).join)
+
+  NUMERIC_REGEX = /(
+    [\+\-]?
+    [\d#{ESCAPED_NON_SPACE_MARKS}][\d#{ESCAPED_MARKS}]*
+  )/ix
+
+  # 1,234,567.89
+  DOT_DECIMAL_REGEX = /\A
+    [\+\-]?
+    (?:
+      (?:\d+)
+      (?:[#{ESCAPED_NON_DOT_MARKS}]\d{3})+
+      (?:\.\d+)?
+    )
+  \z/ix
+
+  # 1.234.567,89
+  COMMA_DECIMAL_REGEX = /\A
+    [\+\-]?
+    (?:
+      (?:\d+)
+      (?:[#{ESCAPED_NON_COMMA_MARKS}]\d{3})+
+      (?:,\d+)?
+    )
+  \z/ix
 
-  def self.parse(input, currency = nil)
-    new.parse(input, currency)
+  # 12,34,567.89
+  INDIAN_NUMERIC_REGEX = /\A
+    [+\-]?
+    (?:
+      (?:\d+)
+      (?:\,\d{2})+
+      (?:\,\d{3})
+      (?:.\d+)?
+    )
+  \z/ix
+
+  # 1,1123,4567.89
+  CHINESE_NUMERIC_REGEX = /\A
+    [+\-]?
+    (?:
+      (?:\d+)
+      (?:\,\d{4})+
+      (?:.\d+)?
+    )
+  \z/ix
+
+  def self.parse(input, currency = nil, options = {})
+    new.parse(input, currency, options)
```

Comment:
I don't think this works, named arguments are not interchangeable with a hash at the end of a method. Try adding a test that uses this class method to know for sure

Reply:
```ruby
[1] pry(main)> class Foo
[1] pry(main)*   def bar(options = {})
[1] pry(main)*     barz(options)
[1] pry(main)*   end
[1] pry(main)*   def barz(a:, b:)
[1] pry(main)*     p [a, b]
[1] pry(main)*   end
[1] pry(main)* end
=> :barz
[2] pry(main)> Foo.new.bar(a: 'it', b: 'works')
["it", "works"]
=> ["it", "works"]
```
I usually prefer this when the method is delegating options. It's also possible to use the `**options` since ruby 2.3. Will add a test that passes an option 👍